hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -146,7 +146,7 @@ protected Option<HoodieRecord<HoodieMetadataPayload>> getRecordByKey(String key,
   @Override
   public List<String> getPartitionPathsWithPrefixes(List<String> prefixes) throws IOException {
     return getAllPartitionPaths().stream()
-        .filter(p -> prefixes.stream().anyMatch(p::startsWith))
+        .filter(p -> prefixes.stream().anyMatch(queryPaths -> p.startsWith(queryPaths + "/") || queryPaths.equals(p)))
```

Comment:
This alters the condition by assuming things such as "/" will be separator and it will be present. Such conditions may not be true for many production scenarios. I think we should not change the condition here. This method is just meant to check for prefixes.

Reply:
This method will return a match partition path in hudi. When this table has a partition like [/inc_day=20221120/opcode=501, /inc_day=20221120/opcode=50, /inc_day=20221120-back/opcode=5000], and the query path prefixes may be a few of the following possibilities:
1) a empty path like "";  
2) a part of partition like "/inc_day=20221120";  
3) a absoulty path like "/inc_day=20221120/opcode=50";

If we just use startWith filter match paths, we will reurn a list with unnecessary partiton paths. Like this:

prefixes="/inc_day=20221120/opcode=50"
matchedPartitionPaths=[/inc_day=20221120/opcode=50, /inc_day=20221120/opcode=501]

or

prefixes="/inc_day=20221120"
matchedPartitionPaths=[/inc_day=20221120/opcode=50, /inc_day=20221120/opcode=501, /inc_day=20221120-back/opcode=5000]

While in most of scenarios the matchedPartitionPaths contains unnecessary partiton paths is right. But in hive will caused a java.lang.RuntimeException: Invalid input path. So we want make the matchedPartitionPaths exclude these unnecessary partiton path.

Reply:
I had changed the condition try to support more scenarios.

This method will return a match partition path in hudi. When this table has a partition like [/inc_day=20221120/opcode=501, /inc_day=20221120/opcode=50, /inc_day=20221120-back/opcode=5000], and the query path prefixes may be a few of the following possibilities:
1) a empty path like "";  
2) a part of partition like "/inc_day=20221120";  
3) a absoulty path like "/inc_day=20221120/opcode=50";

If we just use startWith filter match paths, we will reurn a list with unnecessary partiton paths. Like this:

prefixes="/inc_day=20221120/opcode=50"
matchedPartitionPaths=[/inc_day=20221120/opcode=50, /inc_day=20221120/opcode=501]

or

prefixes="/inc_day=20221120"
matchedPartitionPaths=[/inc_day=20221120/opcode=50, /inc_day=20221120/opcode=501, /inc_day=20221120-back/opcode=5000]

While in most of scenarios the matchedPartitionPaths contains unnecessary partiton paths is right. But in hive will caused a java.lang.RuntimeException: Invalid input path. So we want make the matchedPartitionPaths exclude these unnecessary partiton path.

Reply:
I have changed this condition try to covering all production scenarios. This method we want get all matchedPartitionPaths

When this partition is "/year=2022/month=08/day=30", the query relative partition path prefixes may be these value :
1、a empty path like  "". In this scenarios we should filter all paths return as matched path
2、a prefixes partition path like  "/year=2022";  
3、a absolute path like  "/year=2022/month=08/day=30"

If prefixes is empty ,

Reply:
@onlywangyh : Is the current behavior in Hive differs from Parquet Hive and Presto table ?  If so, can we change in the caller side (for e:g HiveHoodieTableFileIndex) instead of this place.